prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -642,6 +682,27 @@ impl<'tcx, 'vir: 'enc, 'enc> Encoder<'tcx, 'vir, 'enc>
                             unsupported_ty => todo!("unsupported constant literal type: {unsupported_ty:?}"),
                         }
                     }
+                    mir::ConstantKind::Unevaluated(uneval, _) => {
+                        let expr = self
+                            .deps
+                            .require_local::<MirPureEncoder>(MirPureEncoderTask {
+                                encoding_depth: self.encoding_depth + 1,
+                                parent_def_id: uneval.def,
+                                param_env: self.vcx.tcx.param_env(uneval.def),
+                                substs: ty::List::identity_for_item(self.vcx.tcx, uneval.def),
+                                kind: PureKind::Constant(uneval.promoted.unwrap()), 
+                                caller_def_id: self.def_id
+                            })
+                            .unwrap()
+                            .expr;
+
+                        self.vcx.alloc(vir::ExprGenData::Lazy(
+                            vir::vir_format!(self.vcx, "unevaluated const {:?}", uneval.def),
+                            Box::new(move |vcx, lctx: ExprInput<'_>| {
+                                vir::Reify::reify(&expr, vcx, (lctx.0, &[])) //TODO: no idea why i do this so probably wrong
+                            }),
+                        ))
```

Comment:
This looks wrong. The reify call for `expr` should happen first. Its arguments are: the `DefId` for the constant (which is also the `DefId` of its owner) and the arguments to be substituted, so indeed an empty slice for constants. This should not depend on the context for the current function being encoded (even if it sometimes works out because the `DefId`s are the same).

So, `reify` the expression first, then just return it from the `Lazy` directly, without using `lctx`. @JonasAlaif I think we used to have a `lift` function which did this, did we get rid of it at some point?

Reply:
I don't remember removing such a function, so it's probably still around?